File: kyw0716/level 1/햄버거 만들기.js
function solution(ingredient) {
  let answer = 0;
  let stack = [];

  for (let i = 0; i < ingredient.length; i++) {
    stack.push(ingredient[i]);
    if (stack.slice(-4).join("") === "1231") {
      answer += 1;
      for (let i = 0; i < 4; i++) {
        stack.pop();
      }
    }
  }

  return answer;
}

// https://school.programmers.co.kr/learn/courses/30/lessons/133502

File: sk000801/LEVEL1/햄버거 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        for (int i = 0; i < ingredient.length; i++) {
            stack.push(ingredient[i]);
            if (stack.size()>3 && stack.elementAt(stack.size()-1) == 1 && stack.elementAt(stack.size()-2) == 3 && stack.elementAt(stack.size()-3) == 2 && stack.elementAt(stack.size()-4) == 1) {
                answer++;
                stack.pop();
                stack.pop();
                stack.pop();
                stack.pop();
            } 
        }
        return answer;
    }
}

        // StringBuilder ingredients = new StringBuilder();
        // for (int i = 0; i < ingredient.length; i++) {
        //     ingredients.append(ingredient[i]);
        // } 
        // String letter = ingredients.toString();
        // while(letter.contains("1231")) {
        //     letter = letter.replaceAll("1231","");
        // }
        // int answer = (ingredient.length - letter.length()) / 4;
        // return answer;